### PR TITLE
Summarization custom prompt

### PR DESCRIPTION
**Pull request by visahan-24**

### CODE REVIEW

File: assets/training/distillation/components/data_generation/spec.yaml [hunk 1]
```diff
@@ -93,6 +93,12 @@ inputs:
     default: "false"
     description: Enable Chain of density for text summarization
 
+  summary_word_count:
```
[the remark this answers is not in the document]
Comment (visahan-24) [reply assets/training/distillation/components/data_generation/spec.yaml]: @vikasagrawal-ms - renaming it to max_len_summary

File: assets/training/distillation/components/data_generation/spec.yaml [hunk 2]
```diff
@@ -93,6 +93,12 @@ inputs:
     default: "false"
     description: Enable Chain of density for text summarization
 
+  max_len_summary:
```
[the remark this answers is not in the document]
Comment (visahan-24) [reply assets/training/distillation/components/data_generation/spec.yaml:96]: @novaturient95 - The teacher_model_max_new_tokens is applicable for all tasks., Whereas the max_len_summary is specific to summarization. And this is part of cod system prompt template which helps in restricting the size of text summary being generated.

**Review by skasturi — APPROVED** [review]

File: assets/training/distillation/components/data_generation/spec.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (novaturient95) [reply assets/training/distillation/components/data_generation/spec.yaml:96]: Thanks for sharing context on this Visahan. I am good with parameter addn

File: assets/training/distillation/src/validate_pipeline.py [hunk 3]
```diff
@@ -235,6 +241,9 @@ def _validate_record_for_type_NLU_QA(self, record: list) -> str:
         if self._get_cod_status():
             return f"Chain of density is not supported for task type {DataGenerationTaskType.NLU_QUESTION_ANSWERING}"
 
+        if self._get_max_len_summary():
+            return f"Max length summary is not supported for task type {DataGenerationTaskType.NLU_QUESTION_ANSWERING}"
```
Comment (novaturient95) [inline assets/training/distillation/src/validate_pipeline.py:245]: maybe we should not fail run even when this parameter is updated buy the end users. We will simply not use it and if required lets throw a warning

File: assets/training/distillation/src/common/constants.py [hunk 4]
```diff
@@ -70,48 +70,8 @@
...
-  {"Missing_Entities": "<value1>", "Denser_Summary": "<value2>"},
-  {"Missing_Entities": "<value3>", "Denser_Summary": "<value4>"},
-  {"Missing_Entities": "<value5>", "Denser_Summary": "<value6>"},
-  {"Missing_Entities": "<value7>", "Denser_Summary": "<value8>"}
-]
-
-Please ensure that each dense summary should be no more than 80 words.
-"""
+# TEXT SUMMARIZATION DEFAULT OUTPUT WORD COUNT
+DEFAULT_MAX_LEN_SUMMARY = 80
```
Comment (novaturient95) [inline assets/training/distillation/src/common/constants.py:74]: `DEFAULT_MAX_NEW_TOKENS = 128` is way less for summarization as per our discussion. Do you want to update this as well for summarization other than in the notebook only

**Review by novaturient95 — APPROVED** [review]

**Pull request merged**